resource/select_gpu_devices.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+import re
+import dataiku
+from gluonts.mx.context import num_gpus
+import mxnet as mx
+
+def do(payload, config, plugin_config, inputs):
+    """ Retrieve a list of available GPU devices. """
+    choices = []
+
+    if payload.get("parameterName") == "gpu_devices":
+        num_gpu = num_gpus(refresh=True)
```

Comment:
We should do a try/except here in case this fails (https://mxnet.apache.org/versions/master/api/python/docs/api/npx/generated/mxnet.npx.num_gpus.html)

Reply:
the num_gpus() method already does a try/except
```
def num_gpus(refresh=False):
    global NUM_GPUS
    if NUM_GPUS is None or refresh:
        n = 0
        try:
            n = mx.context.num_gpus()
        except mx.base.MXNetError as e:
            logger.error(f"Failure when querying GPU: {e}")
        NUM_GPUS = n
    return NUM_GPUS
```
should we make another try/except ?

Reply:
That's not great, as this hides the error and always returns 0. We should use mx.context.num_gpus() instead of this, and catch the error ourselves, to properly display the "No GPU available case".

Reply:
I put different labels when there is an error with mx.context.num_gpus() ("Fail querying GPUs, please check your CUDA installation") and when there are 0 gpu ("No GPU available, please check your CUDA installation"), what do you think ?

Reply:
Interesting, indeed there could be two cases:
1. The CUDA installation is correct, but there is no GPU on the machine
2. The CUDA installation is not correct, so the GPUs cannot be detected.
Are we sure that we can detect these cases accurately? My concern is that `mx.context.num_gpus()` can return 0 with an incorrect CUDA installation.
If that is the case, then the error messages should say
1. "No GPU detected, please check that your server has GPUs "
2. "No GPU detected, please check your CUDA installation"

Reply:
Actually, there is another case to consider: the GPUs are on a K8S cluster, with DSS hosted on CPU. In that case, the GPUs cannot be detected by the Python select.
Maybe add a button for Kubernetes GPU?